spec.md

```diff
@@ -0,0 +1,228 @@
+# Open Container Initiative
+
+## Artifacts Specification
+
+This Artifacts Spec provides a reference for artifact authors and registry implementors to support new artifact types with pre 1.0 implementations of the [distribution-spec][distribution-spec]. More particularly this repository (spec) has been tasked by the OCI TOB to serve 3 primary goals:
+
+1. **Artifact Authors** - guidance for authoring new artifact types. Including a clearing house for [well known][def-well-known-types] artifact types.
+1. **Registry Operators and Vendors** - guidance for how operators and vendors can support new artifact types, including how they can opt-in or out of well known artifact types. Registry operators that already implement `mediaType` filtering will not have to change. The artifact repo will provide context on how new `mediaTypes` can be used, and how `mediaTypes` can be associated with a type of artifact.
+1. **Clearing House for [Well-known Artifacts][def-well-known-types]** - artifact authors can submit their artifact definitions, providing registry operators a list by which they can easily support.
+
+## Table of Contents
+
+- [Scope](#scope)
+- [Defining OCI Artifact Types](#defining-oci-artifact-types)
+- [Defining a Unique Artifact Type](#defining-a-unique-artifact-type)
+- [Defining Supported Layer Types](#defining-supported-layer-types)
+- [Optional: Defining Config Schema](#optional-defining-config-schema)
+- [Registering Unique Types with IANA](#registering-unique-types-with-iana)
+- [References](#references)
+
+## Scope
+
+Create a first version of the Artifacts specification that is dependent on [OCI manifest][image-manifest] 1.0 to represent individual artifact types including [OCI Image][image-spec], [Helm][helm] and [Singularity][singularity].
+
+**v1.0 of the image-spec** defines index and manifest.
+
+<img src="./media/image-spec-objects.png" width=500x>
+
+**Artifacts 1.0** utilize manifest and index to represent multiple artifact types, including the [OCI Image-spec][image-spec]
+
+<img src="./media/artifacts-spec-objects.png" width=500x>
+
+### Future Scope
+
+Future versions of this specification will support new artifact types representing collections of artifacts using [OCI Index][image-index]. A means to identify an index as a type of artifact will be required.
```

Comment:
I think we should remove any reference to versioning here. Since the primary goal here is guidance and cataloging, I think there would need to be more discussion about what a version of that even means. For example, time based snapshotting of the catalog could be more meaningful than `x.y` versions. Creating a versioned spec was not agreed on when accepted by the TOB (https://github.com/opencontainers/tob/blob/master/proposals/artifacts.md#versioning--roadmap).

Reply:
date based versioning is a reasonable model.. some artifacts might have a different model. Pros/Cons of forcing one versioning model for all artifact types.

Reply:
I think here, though, he's just saying there will be a first artifact spec and then another. Not related to the later allowance for each artifact to have a version.

Reply:
also by spec I took it to mean initial guidance dependent on current manifest.. Then second guidance or update to it based on in flight changes.

Reply:
I suppose you could argue a first set of examples based on the current image spec... then a second set of examples based on the next version of the image spec.. without ever revising the guidance document.. just keep pushing updates.

Reply:
For the versioning, the issue is how to address things we say are supported today:
- manifest based artifacts like Helm, Singularity, OPA
vs. later
- index based artifacts like CNAB or possibly Notary

As we evolved this PR it became clear that we needed a means for registry operators to discern what level of compatibility they support.
If registry-a supports Manifest artifacts, and we add Index, what does that registry operator declare?
When running compliance tests, what are they compliant with and not?
I'm open to suggestions, but what is the proposal to proceed?

Reply:
Initial guidance for using v.x.x.x of the image spec.. Followed by subsequent guidance when using v.x.next.x of the image spec? IOW "provide" examples being guidance for using what we have and what will be available. Center the process guideline more around how to get an artifact through like IANA does.. and leave the version of examples as more a point in time thing as we can add further examples. Should note that the older tools on the prior image spec will likely still be around for a while, even as newer versions of the image spec move that spec forward.

Reply:
> level of compatibility they support

What do you mean by this?

Reply:
> level of compatibility they support
> > What do you mean by this?

I'm referring to this section:
> manifest based artifacts like Helm, Singularity, OPA
> vs. later
> index based artifacts like CNAB or possibly Notary

To support manifest level artifacts, we need:
- oci-manifest 1.0
- A registry operator to open `manifest.config.mediaType` validation. Most, if not all registries restrict `manifest.config.mediaType` to `application/vnd.oci.image.config.v1+json` per: https://github.com/opencontainers/image-spec/blob/master/manifest.md#image-manifest-property-descriptions and the current docker format: `application/vnd.docker.distribution.manifest.v2+json` 

In the future, we plan to add index based artifacts like CNAB.
These will require
- oci-index1.x (a future version that adds `index.config` )
- A registry operator to support oci-index 1.x which will support `index.config.mediaType`. 

The realization was to define a reference by which artifact authors, registry operators and users would know what level of support is required to get different artifact types into a registry. 
If they support Artifacts v1.0, they can push manifest based artifacts. (Helm, Singularity, OPA)
If they support Artifacts v1.1, they can push index based artifacts. (CNAB, ...)

I will try another edit to see if I can get past this simplification

Reply:
@dmcgowan, please see latest version to see if this addresses the concerns.